Keen/Dataset/Datasets.cs

```diff
@@ -0,0 +1,140 @@
+        {
+            if (string.IsNullOrWhiteSpace(datasetName))
+            {
+                throw new KeenException("A dataset name is required.");
+            }
+
+            if (string.IsNullOrWhiteSpace(indexBy))
+            {
+                throw new KeenException("A value to index by is required.");
+            }
+
+            if (string.IsNullOrWhiteSpace(timeframe))
+            {
+                throw new KeenException("A timeframe by is required.");
+            }
+
+            if (string.IsNullOrWhiteSpace(_masterKey))
+            {
+                throw new KeenException("An API masterkey is required to get dataset results.");
+            }
+
+            var datasetResultsUrl = $"{this.GetDatasetUrl(datasetName)}/results";
+
+            var url = $"{datasetResultsUrl}?index_by={indexBy}&timeframe={timeframe}";
+
+            var responseMessage = await _keenHttpClient
+                .GetAsync(url, _masterKey)
+                .ConfigureAwait(continueOnCapturedContext: false);
+
+            var responseString = await responseMessage
+                .Content
+                .ReadAsStringAsync()
+                .ConfigureAwait(continueOnCapturedContext: false);
+
+            var response = JObject.Parse(responseString);
+
+            KeenUtil.CheckApiErrorCode(response);
+
+            if (!responseMessage.IsSuccessStatusCode)
+            {
+                throw new KeenException("Request failed with status: " +
+                                        responseMessage.StatusCode);
+            }
+
+            return response;
+        }
+
+        public async Task<DatasetDefinition> Definition(string datasetName)
+        {
+            if (string.IsNullOrWhiteSpace(datasetName))
+            {
+                throw new KeenException("A dataset name is required.");
+            }
+
+            if (string.IsNullOrWhiteSpace(_masterKey))
+            {
+                throw new KeenException("An API masterkey is required to get dataset results.");
+            }
+
+            var responseMessage = await _keenHttpClient
+                .GetAsync(this.GetDatasetUrl(datasetName), _masterKey)
+                .ConfigureAwait(continueOnCapturedContext: false);
+
+            var responseString = await responseMessage
+                .Content
+                .ReadAsStringAsync()
+                .ConfigureAwait(continueOnCapturedContext: false);
+
+            var response = JObject.Parse(responseString);
+
+            KeenUtil.CheckApiErrorCode(response);
+
+            if (!responseMessage.IsSuccessStatusCode)
+            {
+                throw new KeenException("Request failed with status: " +
+                                        responseMessage.StatusCode);
+            }
+
+            return JsonConvert.DeserializeObject<DatasetDefinition>(responseString, new JsonSerializerSettings { ContractResolver = new SnakeCaseContractResolver()});
```

Comment:
I'll have to think about this and ponder the answers to the questions in my comment above about `SnakeCaseContractResolver` and what it means to always use this resolver for all Dataset definitions.